tests/spark_config_test.py

```diff
@@ -220,6 +224,45 @@ def test_get_mesos_docker_volumes_conf(
         )
         assert sorted(output[validate_key].split(',')) == sorted(set(expected_volumes))
 
+    def test_get_k8s_volume_hostpath_dict(self):
+        assert spark_config._get_k8s_volume_hostpath_dict(
+            '/host/file1', '/container/file1', 'RO', itertools.count(),
+        ) == {
+            'spark.kubernetes.executor.volumes.hostPath.0.mount.path': '/container/file1',
+            'spark.kubernetes.executor.volumes.hostPath.0.options.path': '/host/file1',
+            'spark.kubernetes.executor.volumes.hostPath.0.mount.readOnly': 'true',
+        }
+
+    @pytest.mark.parametrize(
+        'volumes', [
+            None,
+            [
+                {'hostPath': '/host/file1', 'containerPath': '/containter/file1', 'mode': 'RO'},
+                {'hostPath': '/host/file2', 'containerPath': '/containter/file2', 'mode': 'RO'},
+                {'hostPath': '/host/paasta1', 'containerPath': '/container/paasta1', 'mode': 'RO'},
+            ],
+        ],
+    )
+    @pytest.mark.usefixtures('mock_existed_files')
+    def test_get_k8s_docker_volumes_conf(self, volumes):
+        expected_volumes = {}
+
+        _get_k8s_volume = functools.partial(spark_config._get_k8s_volume_hostpath_dict, count=itertools.count())
```

Comment:
Nice usage of partial!

Reply:
it was either this or using contexts, since it's a private method I went with the former.